src/sql/platform/connection/common/utils.ts

```diff
@@ -110,12 +110,19 @@ export function getUriPrefix(ownerUri: string): string {
 }
 
 export function generateUriWithPrefix(connection: IConnectionProfile, prefix: string): string {
-	let id = connection.getOptionsKey();
+	let id = connection.getOptionsKey() + '|id:' + connection.id;
 	let uri = prefix + (id ? id : connection.serverName + ':' + connection.databaseName);
 
 	return uri;
 }
 
+export function generateUriFromExistingUri(connection: IConnectionProfile, id: string): string {
```

Comment:
nit: Name isn't really accurate - this has nothing to do with an existing URI. Something like generateUriFromExistingConnection would be better IMO